### PR TITLE
Fix/better i18n notifications in Async

### PR DESCRIPTION
**Pull request by mzorz**
Coming from https://github.com/wordpress-mobile/WordPress-Android/pull/6713#issuecomment-347926053

This PR eliminates string concatenation where it would present a potential problem for string translation. In replacement, we're using complete different strings for singular and plural wherever possible.

To test:
Follow steps as described in #6713

cc @akirk for looking at strings, and @daniloercoli for code 🙇

### CODE REVIEW

**Comment by daniloercoli** [comment]
I took a quick look and it seems OK, will test it extensively tomorrow, while waiting I18N feedback.

Side note: I like this fine grained notification messages, but do we really need all of them?  Can't just be something like `There was a problem uploading your content. Tap to retry.`

**Comment by mzorz** [comment]
> I took a quick look and it seems OK, will test it extensively tomorrow, while waiting I18N feedback.
Thank you! 🙇 

>  I like this fine grained notification messages, but do we really need all of them? 

We can revisit this in another thread, as the language and all the options for messaging themselves were already discussed/approved in #6713. It was hard to get to an agreement about what the best wording was, that matched all the options. This PR alone only spreads the options in different strings to make them easier to translate.

**Comment by daniloercoli** [comment]
> We can revisit this in another thread, as the language and all the options for messaging themselves were already discussed/approved in #6713. It was hard to get to an agreement about what the best wording was, that matched all the options. This PR alone only spreads the options in different strings to make them easier to translate.

Sure sure, this PR is fine and fixes the problems. I was raising some questions/warnings we should take in consideration for the future. Giving this fine tuning on notification messages brings a lot of work to us, to translators, and reviewers, and not sure it's worth it.

**Comment by daniloercoli** [comment]
@akirk When you have the time, would you mind to take a look at this PR?

**Comment by akirk** [comment]
I already had it on my radar :)

The `file(s)` is problematic for translation. I wonder why we don't do something like: `Uploads remaining: ` or `Uploads failed: ` and append a list of items `1 post` or `%d posts`, `1 page` or `%d pages`, `1 file` or `%d files`.

The notification text could then be `Uploads remaining: 1 post, 2 pages, 4 files`

**Comment by mzorz** [comment]
> The file(s) is problematic for translation. I wonder why we don't do something like: Uploads remaining: or Uploads failed: and append a list of items 1 post or %d posts, 1 page or %d pages, 1 file or %d files.

I decided to address this in the same way we were doing so we can keep the same expressions as defined by Editorial, which also counted in a way that would fit the width of the screen in lower Android versions. Commit 45cb40a shows I expanded the remaining possibilities into separate strings as well.

It adds just a bit of code but nothing that wasn't already along the lines of what we have in this PR.

Hope it looks good now! @akirk @daniloercoli

**Comment by akirk** [comment]
Ok, sorry to rain on your parade but if we want to stick to this format we'll have a problem with `media_file_pages_and_posts_mixed_not_uploaded_one_file` kind of strings. "post" and "page" have different genders in some languages so can't be just used at the same point. But in light of the complication, let's see if translators can cope with it. I just found a typo that I'll comment on the line itself.

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -292,7 +292,27 @@
 
     <!-- post uploads -->
     <string name="upload_failed_param">Upload failed for \"%s\"</string>
-    <string name="media_file_post_mixed_not_uploaded">%1$d %2$s with %3$d files not uploaded</string>
+
+    <string name="media_file_post_singular_mixed_not_uploaded_one_file">1 post with 1 file not uploaded</string>
+    <string name="media_file_posts_plural_mixed_not_uploaded_one_file">%1$d posts with 1 file not uploaded</string>
+    <string name="media_file_page_singular_mixed_not_uploaded_one_file">1 page with 1 file not uploaded</string>
+    <string name="media_file_pages_plural_mixed_not_uploaded_one_file">%1$d pages with 1 file not uploaded</string>
+    <string name="media_file_pages_and_posts_mixed_not_uploaded_one_file">%1$d post / pages with 1 file not uploaded</string>
```
Comment (akirk) [inline WordPress/src/main/res/values/strings.xml]: Shouldn't this be `posts`?

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -292,7 +292,27 @@
...
+
+    <string name="media_file_post_singular_mixed_not_uploaded_files_plural">1 post with %1$d files not uploaded</string>
+    <string name="media_file_posts_plural_mixed_not_uploaded_files_plural">%1$d posts with %2$d files not uploaded</string>
+    <string name="media_file_page_singular_mixed_not_uploaded_files_plural">1 page with %1$d files not uploaded</string>
+    <string name="media_file_pages_plural_mixed_not_uploaded_files_plural">%1$d pages with %2$d files not uploaded</string>
+    <string name="media_file_pages_and_posts_mixed_not_uploaded_files_plural">%1$d post / pages with %2$d files not uploaded</string>
```
Comment (akirk) [inline WordPress/src/main/res/values/strings.xml]: Should probably be `posts`.

File: WordPress/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -292,7 +292,27 @@
...
+
+
+    <string name="media_file_post_singular_only_not_uploaded">1 post not uploaded</string>
+    <string name="media_file_posts_plural_only_not_uploaded">%1$d posts not uploaded</string>
+    <string name="media_file_page_singular_only_not_uploaded">1 page not uploaded</string>
+    <string name="media_file_pages_plural_only_not_uploaded">%1$d pages not uploaded</string>
+    <string name="media_file_pages_and_posts_only_not_uploaded">%1$d post / pages not uploaded</string>
```
Comment (akirk) [inline WordPress/src/main/res/values/strings.xml]: One more instance of `post` that should be `posts`.

File: WordPress/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -292,7 +292,27 @@
...
+
+
```
Comment (akirk) [inline WordPress/src/main/res/values/strings.xml]: Double empty line

**Comment by mzorz** [comment]
addressed [these comments](https://github.com/wordpress-mobile/WordPress-Android/pull/7434#pullrequestreview-102635539) in 0b3d846 @akirk 🙇

**Comment by mzorz** [comment]
> "post" and "page" have different genders in some languages so can't be just used at the same point

I think I can make the distinction and count Pages and Posts separately, but this surely needs another iteration. Another thing that comes to mind is... We're not taking in consideration the width for other languages, so it might be the case that things get translated right, but simply the character count is too long to be shown properly 😱 - it looks like something we'd like to address in a holistic way, maybe running through the whole app with some of the translators to see where things get cut down? Of course this all exceeds this particular PR, but I'm really wondering and hoping to get i18n to be considered from the start on new features, etc, if it makes sense.

**Comment by mzorz** [comment]
just a minor example of what I was saying (unrelated to this particular PR) https://github.com/wordpress-mobile/WordPress-Android/issues/6691 @akirk

**Comment by mzorz** [comment]
thanks for the approval @akirk ! 
ok! ready for last pass @daniloercoli 🙇

**Comment by daniloercoli** [comment]
Tested again, and looks good! :shipit:

**Pull request merged**